build.gradle

```diff
@@ -1,39 +1,79 @@
+import java.nio.file.Files
+import cuchaz.enigma.command.ConvertMappingsCommand
+buildscript {
+    repositories{
+		maven {
+			name "Fabric Repository"
+			url 'https://maven.fabricmc.net'
+		}
+		mavenLocal()
+    }
+
+    dependencies {
+		classpath "cuchaz:enigma:0.14.1-local"
+		classpath "net.fabricmc:stitch:0.2.1-local"
+	}
+}
+
 plugins {
     id 'maven'
     id 'maven-publish'
 }
 
+mkdir("localMappings")
+
 file('mappings').eachFile {
-    if (!it.name.contains(".tiny")) return
+    if(!it.name.endsWith(".tiny")) return
+    File v1MappingFile = it
+    File v2MappingFile = new File(System.getProperty("user.dir") + "/localMappings/${it.name}")
+    if(!v2MappingFile.exists()){
```

Comment:
This should become a task with propper inputs and outputs I think, would ensure the caching is done right, as right now if the v1 changes, the v2 wont be updated.

Reply:
Made it so conversion is done on demand (and is cached properly)